src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -236,6 +215,61 @@ ShouldSyncTableMetadata(Oid relationId)
 }
 
 
+/*
+ * RecreateMetadataSnapshot does the following:
+ *  1. Sets the localGroupId on the worker so the worker knows which tuple in
+ *     pg_dist_node represents itself.
+ *  2. Recreates the distributed metadata on the given worker.
+ *  3. If synchronization succeeds, it sets the pg_dist_node.metadatasynced for
+ *     this node to true.
+ * If raiseOnError is true, it errors out if synchronization fails.
+ */
+bool
+RecreateMetadataSnapshot(WorkerNode *workerNode, bool raiseOnError)
```

Comment:
Maybe SyncMetadataSnapshotToNodes to indicate that this actually sends the metadata as well.

Reply:
done